### PR TITLE
Add CPLD class for Mihawk platform on witherspoon-pfault-analysis 

### PR DESCRIPTION
**Pull request by AndyYFWang**
Add CPLD class for Mihawk platform(2019-10-18)

**This revision is based on the last commit suggestion.** 

If PGOOD signal is abnormal when chassis poweron, read
Mihawk's CPLD-register via I2C to confirm the error.

First confirm whether the PSU is present, then confirm
whether the power_on_error signal is 1 (1 means abnormal).
If the signal is 1, read the error-code-register to
analysis reason.

Tested:
Use command "obmcutil chassiskill" to trigger PGOOD error
action analysis during chassis power on.

### CODE REVIEW

File: org/open_power/Witherspoon/Fault.errors.yaml [hunk 1]
```diff
@@ -24,6 +24,120 @@
 - name: PowerOnFailure
   description: System power failed to turn on
 
+- name: PSUPresentError
+  description: PSU failed to detect
```
Comment (spinler) [inline org/open_power/Witherspoon/Fault.errors.yaml:28]: Does mihawk use policyTable.json?  If so, you'll want to add these errors to it.

File: power-sequencer/mihawk-cpld.hpp [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+    /**
+     * All of powerOnErrorcode are the definition of error-code
+     * which are read on CPLD-error-code-register. 
+     */
+    /**
+     * The definition of error-code:
+     *  
+     */
+    int powerOnErrorCode_0 = 0;
```
Comment (spinler) [inline power-sequencer/mihawk-cpld.hpp:102]: probably not worth changing now, but these should all be const

File: power-sequencer/mihawk-cpld.cpp [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+    // check whether support i2c function
+    unsigned long funcs = 0;
+    if (ioctl(fd, I2C_FUNCS, &funcs) < 0)
+    {
+        std::cerr << "Not support I2C_FUNCS \n";
+        close(fd);
+    }
+
+    // check whether support i2c-read function
```
Comment (spinler) [inline power-sequencer/mihawk-cpld.cpp:360]: but you're doing a write?

Comment (spinler) [reply power-sequencer/mihawk-cpld.cpp:360]: not sure if this matters enough to fix at this point?

File: power-sequencer/mihawk-cpld.hpp (hunk 2 above)
Comment (AndyYFWang) [reply power-sequencer/mihawk-cpld.hpp:102]: Yes, you're right!
I'll add it for this.

File: org/open_power/Witherspoon/Fault.errors.yaml (hunk 1 above)
Comment (AndyYFWang) [reply org/open_power/Witherspoon/Fault.errors.yaml:28]: @spinler 
ok~I will add errors on policyTable.json.

File: power-sequencer/mihawk-cpld.cpp (hunk 3 above)
Comment (AndyYFWang) [reply power-sequencer/mihawk-cpld.cpp:360]: @spinler 
I get what you meant now,
I will fix it.

File: org/open_power/Witherspoon/Fault.errors.yaml (hunk 1 above)
Comment (spinler) [reply org/open_power/Witherspoon/Fault.errors.yaml:28]: Thanks.

File: power-sequencer/mihawk-cpld.cpp (hunk 3 above)
Comment (spinler) [reply power-sequencer/mihawk-cpld.cpp:360]: Do you plan on fixing this here? Or should we just go with what you have?

File: power-sequencer/mihawk-cpld.hpp (hunk 2 above)
Comment (spinler) [reply power-sequencer/mihawk-cpld.hpp:102]: do you plan on changing this in this pull request?

Comment (AndyYFWang) [reply power-sequencer/mihawk-cpld.hpp:102]: Yes, I do.
I will re-push a new request later.

File: power-sequencer/mihawk-cpld.cpp (hunk 3 above)
Comment (AndyYFWang) [reply power-sequencer/mihawk-cpld.cpp:360]: Yes, I do.
I will re-push a new request later.
https://github.com/ibm-openbmc/witherspoon-pfault-analysis/pull/7

File: power-sequencer/mihawk-cpld.hpp (hunk 2 above)
Comment (AndyYFWang) [reply power-sequencer/mihawk-cpld.hpp:102]: https://github.com/ibm-openbmc/witherspoon-pfault-analysis/pull/7

**Comment by rfrandse** [comment]
see https://github.com/ibm-openbmc/witherspoon-pfault-analysis/pull/7

**Pull request closed without merging**